src/NLog/Layouts/Layout.cs

```diff
@@ -63,7 +63,7 @@ public abstract class Layout : ISupportsInitialize, IRenderable
         /// 
         /// Thread-agnostic layouts only use contents of <see cref="LogEventInfo"/> for its output.
         /// </remarks>
-        internal bool ThreadAgnostic { get; private set; }
+        internal bool ThreadAgnostic { get; set; }
```

Comment:
Well internal is "almost" private so it is more restrictive than protected. You are not allowed to upgrade the setter to protected, so it can be accessed by external classes.

Else the entire property should become protected, and part of the official interface.